packages/react/src/components/validation/hooks/use-validation-rule-registration.tsx

```diff
@@ -0,0 +1,18 @@
+import { ValidationRule, ValidationRuleType } from '@devextreme/interim';
+import { useContext, useEffect, useRef } from 'react';
+import { ValidatorContext } from '../contexts/validator-context';
+
+export function useValidationRuleRegistration(
+  type: ValidationRuleType,
+  props: ValidationRule,
+) {
+  const validatorContext = useContext(ValidatorContext);
+  const ruleRegistered = useRef(false);
+  useEffect(() => {
+    if (!ruleRegistered.current) {
+      validatorContext?.registerRule({ ...props, type, reevaluate: true });
+      ruleRegistered.current = true;
+    }
+    // TODO: consider adding rule unregistration on unmount
```

Comment:
Question.

Should we do this ```TODO``` in this PR?

Reply:
Let's make it in a separate PR